### PR TITLE
Add new resource to manage ingest pipelines

### PR DESCRIPTION
**Pull request by olksdr**
It's a part of https://github.com/elastic/terraform-provider-elasticstack/issues/56, which adds the resource to manage the ingest pipelines, where one can provide their own definitions for the processors as a json documents to create and manage the pipeline. 

The  following PRs will be to add `data_source`'s to provide the standard way to create the processors in the declarative way.

### CODE REVIEW

**Review by Crazybus — APPROVED** [review]
LGTM!

**Comment by olksdr** [comment]
@Crazybus could you, please, take another look? I've added a test for this new resource. Kind of forgot about it yesterday 😄

File: internal/elasticsearch/ingest/pipeline_test.go [hunk 1]
```diff
@@ -1 +1,114 @@
...
+}
+
+resource "elasticstack_elasticsearch_ingest_pipeline" "test_pipeline" {
+  name        = "%s"
+  description = "Test Pipeline"
+
+  processors = [
+    jsonencode({
+      set = {
+        description = "My set processor descirption"
```
Comment (Crazybus) [inline internal/elasticsearch/ingest/pipeline_test.go]:
```suggestion
        description = "My set processor description"
```

File: internal/elasticsearch/ingest/pipeline_test.go [hunk 2]
```diff
@@ -1 +1,114 @@
...
```
Comment (Crazybus) [inline internal/elasticsearch/ingest/pipeline_test.go]:
```suggestion
        description = "My set processor description"
```

**Review by Crazybus — APPROVED** [review]
LGTM!

**Comment by olksdr** [comment]
@Crazybus thanks! I've fixed typos. PTAL!

**Review by Crazybus — APPROVED** [review]
LTGM!

**Pull request merged**